workspace/__ardulib__/Graphics/Text.inl

```diff
@@ -0,0 +1,66 @@
+
+Text::Text(XGraphics* parent)
+    : XGraphics(parent) {}
+
+void Text::setPosition(int16_t x, int16_t y) {
+    _pivot = XVector2<int16_t>(x, y);
+}
+
+void Text::setText(XString str) {
+    _str = str;
+}
+
+void Text::setTextColor(XColor* textColor) {
+    _textColor = textColor;
+}
+
+void Text::setTextScale(uint8_t textScale) {
+    _textScale = textScale;
+}
+
+void Text::renderScanline(XRenderer* renderer, int16_t scanline, uint16_t* buffer, size_t bufferSize) {
+
+    if (scanline < _pivot.y || scanline > _pivot.y + 7 * _textScale - 1)
+        return;
+
+    uint16_t color = xcolorTo565(_textColor ? *_textColor : getFGColor());
+
+    uint8_t charCount = 0;
+
+    // For each char in the string.
+    for (auto it = _str.iterate(); it; ++it, charCount++) {
+
+        // Render only ASCII symbols from 32 to 127 else render the "?".
+        uint8_t charNum = ((int)*it < 32 || (int)*it > 127) ? 63 : (int)*it;
```

Comment:
Unreasonable “uint_8t / char / int” type twist in a single expression. All is simpler:

```cpp
char rawChar = *it;
char renderChar = (rawChar < 32 || rawChar > 128) ? '?' : rawChar;
```